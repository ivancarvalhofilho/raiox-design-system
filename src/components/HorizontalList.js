import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const ListContainer = styled.div`
  display: flex;
  position: relative;
  width: 100%;
  padding-top: 16px;
  padding-bottom: 16px;
  overflow-x: hidden;
  font-family: inherit;
`
const List = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
  display: grid;
  gap: 0;
  max-width: 100%;
  grid-auto-flow: column;
  padding-bottom: ${({ scroll }) => scroll ? '16px' : 0};
  overflow-x: auto;

  &::-webkit-scrollbar-track{
	-webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
    background-color: #edeaea;
  }

  &::-webkit-scrollbar-track-piece{
	-webkit-box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.13);
    background-color: #F2F2F2;
  }
  &::-webkit-scrollbar{
    height: 14px;
    background-color: #F2F2F2;
  }
  &::-webkit-scrollbar-thumb{
    border-radius: 4px;
    -webkit-box-shadow: 
      inset 0px 3.5px 0px #F2F2F2,
      inset 0px -3.5px 0px #F2F2F2;
    background-color: rgba(64, 185, 187, 0.4);
  }

  &.active {
    background: rgba(255,255,255,0.3);
    cursor: grabbing;
    cursor: -webkit-grabbing;
    transform: scale(1.02);
  }
`
const ListItem = styled.li`
  display: flex;
  flex-direction: column;
  position: relative;
  width: calc(160px + 96px);
  padding: 2px 48px 12px;
  border-left: 1px solid #E0E0E0;
  box-sizing: border-box;

  &:first-child {
    border-left: none;
    padding-left: 0;
    width: calc(160px + (96px*.5));
  }
`
const Title = styled.p`
  margin: 0;
  color: #595A5C;
  font-size: 13px;
  font-weight: 400;
  line-height: 11px; /* 84.615% */
  letter-spacing: 0.6px;
  margin-bottom: 4px;
`
const Value = styled.p`
  margin: 0;
  color: #303133;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px; /* 157.143% */
  max-width: 160px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-bottom: 8px;
`

/**
 * @typedef {[string,string]} LabelsKey - Array contento na primeira posição a label(título) a ser exibida e segunda posição a chave do objeto com valor
 * @template T
 * @typedef {{
 *  [Property in keyof T]: string | LabelsKey[];
 * } & {labelsKey: LabelsKey[];}} List - Objeto com valores a serem exibidos e seu array de labelsKey
 * @param {{list: List[] | any[]}} props 
 * @example <HorizontalList list={[{
 *    nome: 'Horti Fruti',
      cnpj: "12323132132112",
      codigoFiliacao: "324543552",
      labelsKey: [['Estabelecimento:','nome'],
        ['CNPJ:', 'cnpj'],
        ['Código filiação', 'codigoFiliacao']],
 * }]} />
 * @returns {React.JSX.Element}
 */
const HorizontalList = (props) => {
  const slider = React.useRef(null)
  let isDown = false
  let startX
  let scrollLeft

  const onMouseDown = (e) => {
    isDown = true
    slider.current && slider.current.classList.add('active')
    startX = slider.current && (e.pageX - slider.current.offsetLeft)
    scrollLeft = slider.current && slider.current.scrollLeft
  }

  const onMouseUp = () => {
    isDown = false
    slider.current && slider.current.classList.remove('active')
  }

  const onMouseMove = (e) => {
    if (!isDown) return
    e.preventDefault()
    const x = slider.current && (e.pageX - slider.current.offsetLeft)
    const walk = x && (x - startX) * 1.6
    if (slider.current && walk)
      slider.current.scrollLeft = scrollLeft - walk
  }

  const onMouseLeave = () => {
    isDown = false
    slider.current && slider.current.classList.remove('active')
  }

  return (
    <ListContainer>
      <List
        ref={slider}
        onMouseDown={onMouseDown}
        onMouseUp={onMouseUp}
        onMouseMove={onMouseMove}
        onMouseLeave={onMouseLeave}
        scroll={props.list.length >= 4}
      >
        {
          props.list.map((item, index) => (
            <ListItem key={`estab${index}`}>
              {item.labelsKey.map((key, i) => (
                <div key={`labels${i}`}>
                  <Title>{key[0]}</Title>
                  <Value>{item[key[1]]}</Value>
                </div>
              ))}
            </ListItem>
          ))
        }

      </List>
    </ListContainer>
  )
}

HorizontalList.propTypes = {
  list: PropTypes.array,
}

export { HorizontalList }