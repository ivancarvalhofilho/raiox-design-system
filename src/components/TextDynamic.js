import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import {Tokens} from '../tokens'
import uuidv4, {uuid} from "uuidv4";

const DefaultText = styled.span`
`

const BOLD_COUNTER = 1
const BoldText = styled.span`
  font-weight: ${Tokens.fonts.weight.bold};
`

const ITALIC_COUNTER = 2
const ItalicText = styled.span`
  font-style: italic;
`

const BoldItalicText = styled.span`
  font-style: italic;
  font-weight: ${Tokens.fonts.weight.bold};
`

const TEXT_STYLES_ENUM = {
    0: DefaultText,
    1: BoldText,
    2: ItalicText,
    3: BoldItalicText,
}

const TextDynamic = props => {
    const text = props.children
    const words = text.split(' ')

    return (
        <>
            {
                words.map((word, index) => {
                        let textStyleCounter = 0
                        if (props.boldIndex.includes(index)) {
                            textStyleCounter += BOLD_COUNTER
                        }

                        if (props.italicIndex.includes(index)) {
                            textStyleCounter += ITALIC_COUNTER
                        }

                        const TextStyle = TEXT_STYLES_ENUM[textStyleCounter]

                        return (
                            <DefaultText
                                as={TextStyle}
                                key={uuid()}>{`${word} `}
                            </DefaultText>
                        )
                    }
                )}
        </>
    )
}

export {TextDynamic}

TextDynamic.propTypes = {
    boldIndex: PropTypes.array,
    italicIndex: PropTypes.array,
}
