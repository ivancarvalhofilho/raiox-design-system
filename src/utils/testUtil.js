import { ThemeProvider } from 'styled-components'
import { Tokens } from '../tokens'
import React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'

export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider Tokens={Tokens}>{children}</ThemeProvider>
    ),
  })
}

export function camelize(str) {
  return str.replace(/(?:^\w|[A-Z]|\b\w)/g, function(word, index) {
    return index === 0 ? word.toLowerCase() : word.toUpperCase();
  }).replace(/\s+/g, '').replace(/\W/gm,'');
}
