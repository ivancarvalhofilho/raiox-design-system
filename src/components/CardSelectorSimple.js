import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import {Tokens} from '../tokens'
import {fontStyleMaker} from '../utils'
import {Icon} from "./Icon";


const CardSelectorSimpleStyle = styled.div`
  transition: 0.15s;
  background: ${Tokens.colors.neutral.light.base};
  box-shadow: 0 0 0 1px ${Tokens.colors.neutral.dark["03"]};
  border-radius: ${Tokens.border.radius.sm};
  user-select: none;
  cursor: pointer;

  pointer-events: ${props => props.disabled && 'none'};
  opacity: ${props => props.disabled && Tokens.opacity.level.medium};
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 8px;
  margin-top: 8px;
`

const Title = styled.span`
  ${fontStyleMaker({
    fontFamily: "body",
    fontWeight: "regular",
    fontSize:"xs"
  })};
  color: ${Tokens.colors.brand.secondary.dark};
`
const CardSelectorSimple = props => (
    <CardSelectorSimpleStyle
        disabled={props.disabled}
        onClick={() => props.onClick()}
        style={props.style}
        className={props.className}
    >
        <Title>{props.title}</Title>
        {props.arrowIcon &&
        <Icon path={Tokens.icons["single-arrow-right"]} size='md'/>
        }
        {(!props.arrowIcon && props.rightIcon) && (props.rightIcon)}
    </CardSelectorSimpleStyle>
)

CardSelectorSimple.defaultProps = {
  arrowIcon: true,
}

CardSelectorSimple.propTypes = {
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  title: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
  ]),
  style: PropTypes.any,
  className: PropTypes.any,
  arrowIcon: PropTypes.bool,
  rightIcon: PropTypes.element,
}

export {CardSelectorSimple}