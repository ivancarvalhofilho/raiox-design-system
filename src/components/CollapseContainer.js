import PropTypes from 'prop-types'
import React, {useRef, useState} from 'react'
import {Icon} from './Icon'
import styled from 'styled-components'
import {UnmountClosed} from 'react-collapse'
import {
    Container as StyledContainer,
    Divider as DividerStyled,
} from './StyledComponents'
import {Tokens} from "../tokens";
import {fontStyleMaker, handleOutsideDivClick} from "../utils";

const Container = styled(StyledContainer)`
  box-shadow: ${Tokens.shadow.level0};
  padding: 0;
  background: white;
  margin-top: ${Tokens.spacing.inline.xxxs};
  margin-bottom: ${Tokens.spacing.inline.xxxs};
`
const Header = styled.div`
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: ${Tokens.spacing.inset.xs};
`
const Button = styled.div`
  display: flex;
  padding-left: ${Tokens.spacing.inline.nano};
`

const Divider = styled(DividerStyled)`
  width: 90%;
  margin: auto;
  position: relative;
`
const Body = styled.div`
  ${fontStyleMaker({
    fontFamily: "body",
    fontWeight: "regular",
    fontSize: "xs"
  })};
  line-height: 21px;
  color: ${Tokens.colors.neutral.dark.base};
  padding: ${Tokens.spacing.inset.xs};
`
const Title = styled.span`
  ${fontStyleMaker({
    fontFamily: "body",
    fontWeight: "regular",
    fontSize: "xs",
  })};
  line-height: 21px;
  color: ${Tokens.colors.brand.secondary.dark};
`
const CollapseContainer = props => {
    const [opened, setOpened] = useState(false)

    const openRef = useRef()
    openRef.current = opened

    const wrapperRef = handleOutsideDivClick(
        () => {
            if (openRef.current){
                setTimeout(() => {
                    setOpened(false)
                }, 50)
            }
        },
    )

    return (
        <Container ref={wrapperRef}>
            <Header id="header" onClick={() => setOpened(!opened)}>
                <Title>{props.title}</Title>
                <Button>
                    <Icon
                        path={Tokens.icons['chevron-down']}
                        size="16px"
                        rotate={opened ? 180 : 0}
                        appearance={opened ? 'primary' : 'default'}
                    />
                </Button>
            </Header>
            <UnmountClosed isOpened={opened} style={{padding: 0}}>
                <Divider horizontal/>
                <Body id="children">{props.children}</Body>
            </UnmountClosed>
        </Container>
    )
}

export {CollapseContainer}
CollapseContainer.propTypes = {
    children: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
    title: PropTypes.string,
}
