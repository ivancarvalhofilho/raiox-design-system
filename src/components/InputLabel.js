import PropTypes from 'prop-types'
import React, {useRef, useState} from 'react'
import styled from 'styled-components'
import {Tokens} from '../tokens'
import {fontStyleMaker} from '../utils/FontUtil'
import {Icon} from "./Icon";

const SuspendedLabel = styled.div`
  position: absolute;
  z-index: 1;
  margin: 12px;
  padding: 4px;
  color: ${Tokens.colors.neutral.dark['01']};
  ${fontStyleMaker({
	fontFamily: 'body',
	fontWeight: 'regular',
	fontSize: 'xs',
})};
  background: inherit;
  transition: 0.25s;
  top: 0;
  pointer-events: none;
  user-select: none;
  ${(props) => props.hasText && makeLabelSuspended()}
`

function makeLabelSuspended() {
	return {
		top: '-22px',
		...fontStyleMaker({
			fontFamily: 'body',
			fontWeight: 'regular',
			fontSize: 'xxs',
		}),
	}
}

const InputLabelStyled = styled.input`
  padding: 16px;
  box-shadow: inset 0 0 0 1px ${Tokens.colors.neutral.dark['03']};
  border: none;
  box-sizing: border-box;
  border-radius: 4px;
  background: ${Tokens.colors.neutral.light.base};
  transition: 0.25s;
  color: ${Tokens.colors.neutral.dark.base};
  ${fontStyleMaker({
	fontFamily: 'body',
	fontWeight: 'regular',
	fontSize: 'xs',
})};
  width: inherit;
  &:focus {
    outline: none;
    box-shadow: inset 0 0 0 2px ${Tokens.colors.brand.primary.darkest};
  }

  &:focus + div {
    ${makeLabelSuspended()};
  }
  ${props => props.isPasswordType && `
    padding-right: 100px;
  `}
  ${(props) =>
	props.hasError &&
	`
		box-shadow: inset 0 0 0 2px ${Tokens.colors.feedback.danger.dark} !important;
	`}

  &:-webkit-autofill {
    -webkit-background-clip: text;
  }
`
const InputLabelContainer = styled.div`
  background: ${Tokens.colors.neutral.light.base};
  ${(props) =>
	props.disabled &&
	`
    opacity:${Tokens.opacity.level.medium};
    pointer-events: none;
  `};
  position: relative;
  width: 100%;
`
const SpanError = styled.span`
  position: relative;
  display: flex;
  color: ${Tokens.colors.feedback.danger.dark};
  ${fontStyleMaker({
	fontWeight: 'regular',
	fontFamily: 'body',
	fontSize: 'xs',
})};
  margin-top: ${Tokens.spacing.inline.nano};
	text-align: start;
`
const PasswordEyeContainer = styled.div`
  display: flex;
  top: 0;
  right: 0;
  position: absolute;
  align-items: center;
  justify-content: center;
  & > span {
    ${fontStyleMaker({
	fontFamily: "body",
	fontSize: "xxs",
	fontWeight: "regular"
})};
    margin-left: ${Tokens.spacing.stack.nano};
    color: ${Tokens.colors.neutral.dark["01"]};
  }
  height: 48px;
  margin-right: ${Tokens.spacing.stack.xxxs};

  cursor: pointer;
  user-select: none;
`
const InputLabel = (props) => {
	const [visiblePassword, setVisiblePassword] = useState(false)

	let inputType = props.inputType || 'text'

	if (inputType === 'password') {
		inputType = visiblePassword ? 'text' : 'password'
	}

	const isPasswordType = props.inputType === 'password';

	function camelize(str) {
		return str.replace(/(?:^\w|[A-Z]|\b\w)/g, function(word, index) {
			return index === 0 ? word.toLowerCase() : word.toUpperCase();
		}).replace(/\s+/g, '');
	}

	function getTestID() {
		return props.testID || (
			camelize(props.label).replace(/\W/gm,'')
		);
	}

	return (
		<InputLabelContainer
			disabled={props.disabled}
			style={props.style}
			className={props.className}
		>
			<InputLabelStyled
				type={inputType}
				value={props.text}
				hasError={props.error}
				maxLength={props.maxLength}
				onChange={(text) => {
					props.setText(text.target.value)
				}}
				autoFocus={props.autofocus}
				name={props.label}
				id={props.label}
				onKeyDown={(event) => {
					if (!!props.submitOnEnter && event.key === 'Enter') {
						props.submitOnEnter()
					}
				}}
				onBlur={props.onBlur}
				isPasswordType={isPasswordType}
				testID={getTestID()}
			/>
			<SuspendedLabel hasText={!!props.text}>{props.label}</SuspendedLabel>
			{props.error && <SpanError>{props.errorLabel}</SpanError>}
			{isPasswordType && <PasswordEyeContainer onClick={() => setVisiblePassword(!visiblePassword)}>
				<Icon path={visiblePassword ? Tokens.icons["eye-closed"] : Tokens.icons.eye}
							appearance={props.error ? 'danger' : 'primary'} size={20}/>
				<span>{visiblePassword ? 'Esconder' : 'Mostrar'}</span>
			</PasswordEyeContainer>}
		</InputLabelContainer>
	)
}

export {InputLabel}
InputLabel.propTypes = {
	state: PropTypes.string,
	text: PropTypes.string,
	setText: PropTypes.func,
	onBlur: PropTypes.func,
	maxLength: PropTypes.number,
	submitOnEnter: PropTypes.func,
	disabled: PropTypes.bool,
	error: PropTypes.bool,
	errorLabel: PropTypes.string,
	inputType: PropTypes.string,
	autofocus: PropTypes.bool,
	testID: PropTypes.string,
}
