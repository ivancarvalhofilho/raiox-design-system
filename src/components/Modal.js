import PropTypes from 'prop-types'
import React, { useEffect, useRef } from 'react'
import styled from 'styled-components'
import { Icon } from './Icon'
import { Tokens } from '../tokens'
import { handleOutsideDivClick } from '../utils'

const BackgroundContainer = styled.div`
  position: fixed;
  width: 100%;
  z-index: 1000;
  height: 100%;
  background: rgba(0, 0, 0, 0.75);
  top: 0;
  overflow: hidden;
  left: 0;
  display: ${(props) => (props.show ? 'flex' : 'none')};
  align-items: center;
  justify-content: center;
`
const ModalHeader = styled.div`
  display: flex;
  justify-content: space-between;
`
const Title = styled.div``
const ModalContent = styled.div`
  ${(props) => 
    !props.ignoreInsidePadding && `padding: ${Tokens.spacing.inline.xxxs};`}
  box-sizing: border-box;
  height: 100%;
`
const ModalContainer = styled.div`
  box-shadow: 0px 0px 8px rgba(0, 39, 64, 0.1);
  border-radius: 5px;
  ${(props) => 
    !props.ignoreHeight && `height: ${props.height ? `${props.height}px` : '100%'};`}
  width: ${(props) => (props.width ? `${props.width}px` : '520px')};
  transition: 0.5s;
  background: white;
  position: absolute;
  padding: ${Tokens.spacing.inline.xxxs};

  animation: modalSpawnAnimation 0.5s;
  animation-iteration-count: 1;
`
const Modal = (props) => {
  const openRef = useRef()
  openRef.current = props.show

  const modalBodyRef = handleOutsideDivClick(() => {
    if (openRef.current) {
      props.onClickOut(false)
    }
  })

  function onKeyboardHit(event) {
    if (event.key === 'Escape') {
      props.onClose()
    }
  }

  useEffect(() => {
    document.addEventListener('keydown', onKeyboardHit)
    return () => {
      document.removeEventListener('keydown', onKeyboardHit)
    }
  }, [])

  return (
    <BackgroundContainer
      show={props.show}
      style={props.style}
      className={props.className}
    >
      <ModalContainer
        height={props.height}
        width={props.width}
        ref={modalBodyRef}
        ignoreHeight={props.ignoreHeight}
      >
        <ModalHeader>
          <Title>{props.title}</Title>
          {props.closable && (
            <Icon
              size="16px"
              onClick={props.onClose}
              path={Tokens.icons.close}
              style={{
                position: 'absolute',
                right: '16px',
                top: '16px',
              }}
            />
          )}
        </ModalHeader>
        <ModalContent ignoreInsidePadding={props.ignoreInsidePadding}>{props.children}</ModalContent>
      </ModalContainer>
    </BackgroundContainer>
  )
}

export { Modal }
Modal.propTypes = {
  children: PropTypes.any,
  style: PropTypes.any,
  className: PropTypes.any,
  closable: PropTypes.bool,
  onClickOut: PropTypes.func,
  onClose: PropTypes.func,
  show: PropTypes.bool,
  title: PropTypes.string,
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  width: PropTypes.number,
  ignoreHeight: PropTypes.bool,
  ignoreInsidePadding: PropTypes.bool,
}
