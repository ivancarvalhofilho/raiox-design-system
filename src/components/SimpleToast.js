import PropTypes from 'prop-types'
import React, { useState } from 'react'
import styled from 'styled-components'
import { Tokens } from '../tokens'
import { Icon } from "./Icon";
import { fontStyleMaker } from "../utils/FontUtil";

const SimpleToastStyle = styled.div`
	position: relative;
	display: flex;
	justify-content: center;
	align-items: center;
	background: ${props => props.background};	
	padding: ${Tokens.spacing.inset.xs};
	width: fit-content;
	cursor: pointer;
	min-width: 200px;
	transition: 1s;
	opacity: ${props => props.isVisible ? 1 : 0};
	pointer-events: ${props => props.isVisible ? 'all' : 'none'};
	& > span {
		${fontStyleMaker({
	fontSize: "xs",
	fontFamily: "head",
	fontWeight: "medium"
})};
	}
	color: ${props => props.fontColor};

  	opacity: ${props => props.disabled && Tokens.opacity.level.medium};
`

const ToastProgressBar = styled.div`
	background: white;
	opacity: 0.5;
	position: absolute;
	height: 5px;
	bottom: 0;
	left: 0;
	width: 100%;
	${props => props.visible && `animation: simpleToastProgressBar ${props.lifetime}ms;`}
	animation-iteration-count: 1;
	animation-timing-function: cubic-bezier(0, 0, 0.2, 1);
`

const toastStyles = {
	'success': {
		icon: Tokens.icons['check-rounded'],
		appearance: 'light',
		fontColor: Tokens.colors.neutral.light.base,
		borderColor: Tokens.colors.feedback.success.darkest,
		backgroundColor: Tokens.colors.feedback.success.darkest,
	},
	'info': {
		icon: Tokens.icons['info'],
		appearance: 'light',
		fontColor: Tokens.colors.neutral.light.base,
		borderColor: Tokens.colors.feedback.info.darkest,
		backgroundColor: Tokens.colors.feedback.info.darkest,
	},
	'warning': {
		icon: Tokens.icons['warning-triangle'],
		appearance: 'dark',
		fontColor: Tokens.colors.neutral.dark.base,
		borderColor: Tokens.colors.feedback.warning.dark,
		backgroundColor: Tokens.colors.feedback.warning.dark,
	},
	'error': {
		icon: Tokens.icons['close-rounded'],
		appearance: 'light',
		fontColor: Tokens.colors.neutral.light.base,
		borderColor: Tokens.colors.feedback.danger.darkest,
		backgroundColor: Tokens.colors.feedback.danger.darkest,
	}
}

const SimpleToast = props => {
	const [timeoutId, setTimeoutId] = useState(0)

	if (props.isVisible && props.setInvisible && timeoutId == 0) {
		const timeoutIdTemp = setTimeout(() => { props.setInvisible(false) }, props.lifetime)
		setTimeoutId(timeoutIdTemp)
	} else if (!props.isVisible && !!timeoutId) {
		clearTimeout(timeoutId)
		setTimeoutId(0)
	}

	return (
		<SimpleToastStyle
			isVisible={props.isVisible}
			background={toastStyles[props.type].backgroundColor}
			borderColor={toastStyles[props.type].borderColor}
			fontColor={toastStyles[props.type].fontColor}
			disabled={props.disabled}
			onClick={() => {
				props.onClick ? props.onClick() : props.setInvisible?.call(false)
				clearTimeout(timeoutId)
			}}
			style={props.style}>
			<Icon path={toastStyles[props.type].icon} appearance={toastStyles[props.type].appearance} size={22}
				style={{ marginRight: Tokens.spacing.stack.nano }} />
			<span style={{ flex: 1, position: "relative", top: '1px' }}> {props.text} </span>
			<Icon path={Tokens.icons["close"]} size={14} appearance={toastStyles[props.type].appearance} style={{ marginLeft: Tokens.spacing.stack.nano }} />
			<ToastProgressBar
				visible={props.isVisible}
				lifetime={props.lifetime}
			/>
		</SimpleToastStyle>
	)
}

export { SimpleToast }
SimpleToast.propTypes = {
	identifier: PropTypes.string,
	isVisible: PropTypes.bool,
	setInvisible: PropTypes.func,
	type: PropTypes.string,
	onClick: PropTypes.func,
	disabled: PropTypes.bool,
	text: PropTypes.string,
	appearance: PropTypes.string,
	lifetime: PropTypes.number,
	style: PropTypes.object,
}
