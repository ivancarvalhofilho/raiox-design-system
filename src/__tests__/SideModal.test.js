import { mountWithTheme } from '../utils/testUtil'
import toJson from 'enzyme-to-json'
import React from 'react'
import { SideModal } from '../components'

it('should render', () => {
  const spy = jest.spyOn(global.console, 'error')
  const props = {
    closable: true,
    show: true,
    title: 'test',
  }

  const app = mountWithTheme(<SideModal {...props}>TestChildren</SideModal>)
  expect(toJson(app)).toMatchSnapshot()
  expect(spy).not.toHaveBeenCalled()
})

it('should open', () => {
  const mockClick = jest.fn()
  let props = {
    closable: true,
    show: false,
    title: 'test',
  }

  const modal = mountWithTheme(
    <SideModal
      {...props}
      onClickOut={() => {
        mockClick()
        props = { ...props, show: !props.show }
      }}
    >
      TestChildren
    </SideModal>,
  )
  jest.useFakeTimers()
  const sideModalContainer = modal.find('#sideModalContainer')
  sideModalContainer.first().simulate('click')
  expect(mockClick).not.toHaveBeenCalled()
  expect(modal).toHaveStyleRule('display', 'none')
  props = { ...props, show: true }
  modal.setProps(props)
  modal.update()
  jest.advanceTimersByTime(600)
  expect(modal).toHaveStyleRule('display', 'flex')
  modal.simulate('click')
  modal.setProps(props)
  jest.advanceTimersByTime(600)
  modal.update()
  expect(modal).toHaveStyleRule('display', 'none')
  expect(mockClick).toHaveBeenCalled()
})
