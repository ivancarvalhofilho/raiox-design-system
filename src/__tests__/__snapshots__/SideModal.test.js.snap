// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transform: rotate(-deg);
  -ms-transform: rotate(-deg);
  transform: rotate(-deg);
}

.c0 {
  position: absolute;
  z-index: 1000;
  width: 100%;
  height: 100%;
  overflow: hidden;
  background: rgba(0,0,0,0.75);
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  padding-bottom: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c4 {
  font-family: Mark Pro;
  font-weight: 700;
  font-size: 18px;
  color: #003A60;
  width: 80%;
  text-align: center;
  margin: 0 auto;
}

.c1 {
  box-shadow: 0px 0px 8px rgba(0,39,64,0.1);
  border-radius: 5px;
  width: 374px;
  overflow: auto;
  background: white;
  -webkit-transition: right 0.5s;
  transition: right 0.5s;
  position: absolute;
  right: 0px;
  padding: 17px;
  height: 100%;
}

<SideModal
  closable={true}
  show={true}
  title="test"
>
  <styled.div
    onClick={[Function]}
    show={true}
  >
    <div
      className="c0"
      onClick={[Function]}
    >
      <styled.div
        id="sideModalContainer"
        onClick={[Function]}
        show={true}
        width={0}
      >
        <div
          className="c1"
          id="sideModalContainer"
          onClick={[Function]}
          width={0}
        >
          <styled.div>
            <div
              className="c2"
            >
              <Icon
                path="<svg width=\\"14\\" height=\\"14\\" viewBox=\\"0 0 14 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M14 1.4094L12.5906 0L7 5.5906L1.4094 0L0 1.4094L5.5906 7L0 12.5906L1.4094 14L7 8.4094L12.5906 14L14 12.5906L8.4094 7L14 1.4094Z\\" fill=\\"#979899\\"/></svg>"
                size="16px"
              >
                <Styled(SVGInline)
                  fill="#979899"
                  path="<svg width=\\"14\\" height=\\"14\\" viewBox=\\"0 0 14 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M14 1.4094L12.5906 0L7 5.5906L1.4094 0L0 1.4094L5.5906 7L0 12.5906L1.4094 14L7 8.4094L12.5906 14L14 12.5906L8.4094 7L14 1.4094Z\\" fill=\\"#979899\\"/></svg>"
                  size="16px"
                  style={Object {}}
                  svg="<svg width=\\"14\\" height=\\"14\\" viewBox=\\"0 0 14 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M14 1.4094L12.5906 0L7 5.5906L1.4094 0L0 1.4094L5.5906 7L0 12.5906L1.4094 14L7 8.4094L12.5906 14L14 12.5906L8.4094 7L14 1.4094Z\\" fill=\\"#979899\\"/></svg>"
                  width="16px"
                >
                  <SVGInline
                    className="c3"
                    classSuffix="-svg"
                    cleanup={Array []}
                    cleanupExceptions={Array []}
                    component="span"
                    fill="#979899"
                    path="<svg width=\\"14\\" height=\\"14\\" viewBox=\\"0 0 14 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M14 1.4094L12.5906 0L7 5.5906L1.4094 0L0 1.4094L5.5906 7L0 12.5906L1.4094 14L7 8.4094L12.5906 14L14 12.5906L8.4094 7L14 1.4094Z\\" fill=\\"#979899\\"/></svg>"
                    size="16px"
                    style={Object {}}
                    svg="<svg width=\\"14\\" height=\\"14\\" viewBox=\\"0 0 14 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M14 1.4094L12.5906 0L7 5.5906L1.4094 0L0 1.4094L5.5906 7L0 12.5906L1.4094 14L7 8.4094L12.5906 14L14 12.5906L8.4094 7L14 1.4094Z\\" fill=\\"#979899\\"/></svg>"
                    width="16px"
                  >
                    <span
                      className="SVGInline c3"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<svg class=\\"SVGInline-svg sc-bdfBwQ-svg c3-svg\\" fill=\\"#979899\\" style=\\"width: 16px;height: 16px;\\" width=\\"14\\" height=\\"14\\" viewBox=\\"0 0 14 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M14 1.4094L12.5906 0L7 5.5906L1.4094 0L0 1.4094L5.5906 7L0 12.5906L1.4094 14L7 8.4094L12.5906 14L14 12.5906L8.4094 7L14 1.4094Z\\" fill=\\"#979899\\"/></svg>",
                        }
                      }
                      path="<svg width=\\"14\\" height=\\"14\\" viewBox=\\"0 0 14 14\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M14 1.4094L12.5906 0L7 5.5906L1.4094 0L0 1.4094L5.5906 7L0 12.5906L1.4094 14L7 8.4094L12.5906 14L14 12.5906L8.4094 7L14 1.4094Z\\" fill=\\"#979899\\"/></svg>"
                      size="16px"
                      style={Object {}}
                    />
                  </SVGInline>
                </Styled(SVGInline)>
              </Icon>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c4"
            >
              test
            </div>
          </styled.div>
          <styled.div>
            <div
              className=""
            >
              TestChildren
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</SideModal>
`;
